pkg/services/accesscontrol/acimpl/metrics.go

```diff
@@ -0,0 +1,46 @@
+package acimpl
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/grafana/pkg/infra/metrics/metricutil"
+)
+
+const exporterName = "grafana"
+
+var (
+	// mAccessEngineEvaluationsSeconds is a summary for evaluating access for a specific engine (RBAC and zanzana)
+	mAccessEngineEvaluationsSeconds *prometheus.HistogramVec
+
+	// mZanzanaEvaluationStatusTotal is a metric for zanzana evaluation status
+	mZanzanaEvaluationStatusTotal *prometheus.CounterVec
+)
+
+var once sync.Once
+
+func initMetrics() {
+	once.Do(func() {
+		mAccessEngineEvaluationsSeconds = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:      "access_engine_evaluations_seconds",
+			Help:      "Histogram for evaluation time for the specific access control engine (RBAC and zanzana).",
+			Namespace: exporterName,
+			Buckets:   prometheus.ExponentialBuckets(0.00001, 4, 10),
+		},
+			[]string{"engine"},
+		)
+
+		mZanzanaEvaluationStatusTotal = metricutil.NewCounterVecStartingAtZero(
+			prometheus.CounterOpts{
+				Name:      "zanzana_evaluation_status_total",
+				Help:      "evaluation status (success or error) for zanzana",
+				Namespace: exporterName,
+			}, []string{"status"}, map[string][]string{"status": {"success", "error"}})
+
+		prometheus.MustRegister(
```

Comment:
am pretty sure we are supposed to use inject ` prometheus.Registerer` through wire. Similiar to what we have done in https://github.com/grafana/grafana/blob/main/pkg/services/authn/authnimpl/service.go#L47-L50. 
 
Also we should only register these metrics if feature toggle is turned on.

Reply:
I also have a preference to store metrics in a struct instead of package level variables, see https://github.com/grafana/grafana/blob/main/pkg/services/authn/authnimpl/metrics.go as an example. But it is just a preference I am okey with either

Reply:
Yes, we could use prometheus register, but I didn't want to inject new wire dependency :) I'll rewrite my PR to use it

Reply:
Its probably fine for now because it is behind feature toggle and we are not ready to use it yet anyway. You can also leave a todo and we can merge as is :)